src/par_iter/test.rs

```diff
@@ -272,6 +272,144 @@ pub fn check_range_exact_and_bounded() {
 }
 
 #[test]
+pub fn check_cmp() {
+    let a: Vec<usize> = (0..1024).collect();
+    let b: Vec<usize> = (0..1024).collect();
+
+    let result = a.cmp(&b);
+
+    assert!(result == ::std::cmp::Ordering::Equal);
+}
+
+#[test]
+pub fn check_cmp_lt() {
+    let a: Vec<usize> = (0..1024).collect();
+    let b: Vec<usize> = (1..1024).collect();
+
+    let result = a.cmp(&b);
+
+    assert!(result == ::std::cmp::Ordering::Less);
+}
+
+#[test]
+pub fn check_cmp_gt() {
+    let a: Vec<usize> = (1..1024).collect();
+    let b: Vec<usize> = (0..1024).collect();
+
+    let result = a.cmp(&b);
+
+    assert!(result == ::std::cmp::Ordering::Greater);
+}
+
+#[test]
+pub fn check_partial_cmp() {
+    let a: Vec<usize> = (0..1024).collect();
+    let b: Vec<usize> = (0..1024).collect();
+
+    let result = a.partial_cmp(&b);
+
+    assert!(result == Some(::std::cmp::Ordering::Equal));
+}
+
+#[test]
+pub fn check_partial_cmp_lt() {
+    let a: Vec<usize> = (0..1024).collect();
+    let b: Vec<usize> = (1..1024).collect();
+
+    let result = a.partial_cmp(&b);
+
+    assert!(result == Some(::std::cmp::Ordering::Less));
+}
+
+#[test]
+pub fn check_partial_cmp_gt() {
+    let a: Vec<usize> = (1..1024).collect();
+    let b: Vec<usize> = (0..1024).collect();
+
+    let result = a.partial_cmp(&b);
+
+    assert!(result == Some(::std::cmp::Ordering::Greater));
+}
+
+#[test]
+pub fn check_partial_cmp_none() {
+    let mut a: Vec<f32> = (0..1024u16).map(From::from).collect();
+    let b: Vec<f32> = (0..1024u16).map(From::from).collect();
+
+    a[0] = ::std::f32::NAN;
+
+    let result = a.partial_cmp(&b);
+
+    assert!(result == None);
+}
+
+#[test]
+pub fn check_eq() {
+    let a: Vec<usize> = (0..1024).collect();
+    let b: Vec<usize> = (0..1024).collect();
+
+    let result = a.eq(&b);
+
+    assert!(result);
+}
+
+#[test]
+pub fn check_ne() {
+    let a: Vec<usize> = (0..1024).collect();
+    let b: Vec<usize> = (1..1024).collect();
+
+    let result = a.ne(&b);
+
+    assert!(result);
+}
+
+#[test]
+pub fn check_lt() {
+    let a: Vec<usize> = (0..1024).collect();
+    let b: Vec<usize> = (1..1024).collect();
+
+    let result = a.lt(&b);
+
+    assert!(result);
+}
+
+#[test]
+pub fn check_le() {
+    let a: Vec<usize> = (0..1024).collect();
+    let b: Vec<usize> = (1..1024).collect();
+    let c: Vec<usize> = (0..1024).collect();
+
+    let result_lt = a.le(&b);
+    let result_eq = a.le(&c);
+
+    assert!(result_lt);
+    assert!(result_eq);
+}
+
+#[test]
+pub fn check_gt() {
+    let a: Vec<usize> = (1..1024).collect();
+    let b: Vec<usize> = (0..1024).collect();
+
+    let result = a.gt(&b);
+
+    assert!(result);
+}
+
+#[test]
+pub fn check_ge() {
+    let a: Vec<usize> = (1..1024).collect();
+    let b: Vec<usize> = (0..1024).collect();
+    let c: Vec<usize> = (1..1024).collect();
+
+    let result_gt = a.ge(&b);
+    let result_eq = a.ge(&c);
+
+    assert!(result_gt);
+    assert!(result_eq);
+}
```

Comment:
can you also add some tests involving floats (which are not fully ordered)? 

http://play.integer32.com/?gist=87daf4489fb2ca744e1ea99abac0f443&version=stable would be a good one, if we added the parallel version and compared them against one another.

It'd also be good to add a test with some randomly generated data, using a fixed seed. 

Something like this:

```rust
#[test]
pub fn cmp_and_partial_cmp() {
    use rand::{Rng, SeedableRng, XorShiftRng};

    let mut rng = XorShiftRng::from_seed([14159, 26535, 89793, 23846]);
    let a: Vec<i32> = rng.gen_iter().take(1024).collect();
    let b: Vec<i32> = rng.gen_iter().take(1024).collect();
    for i in 0..a.len() {
        assert_eq!(a[i..].par_iter().partial_cmp(b[i..]));
        assert_eq!(a[i..].iter().cmp(b[i..]);
    }
}
```

(a twist on [@utkarshkukreti's test](https://github.com/nikomatsakis/rayon/pull/159#issuecomment-261599932))